### ISSUE TITLE
[Feature Request] Allow a configuration object to be passed to linter.lint() instead of using Configuration.findConfiguration

### ISSUE THREAD

**Issue by saadq** [palantir/tslint#3850]
This isn't a bug report, so I erased the issue template – hope that's okay.

For my use case, I am using [TSLint as a Node.js library](https://palantir.github.io/tslint/usage/library/), and I need to create the TSLint config dynamically based on user input. Currently, I do this by dynamically creating a `tslint.json` file, and then using `Configuration.findConfiguration()` to get that config and then finally using that with `linter.lint()`.

It would be a lot easier to use TSLint programmatically if one could just simply pass a regular config object to linter.lint that follows the TSLint schema. For example, something like:

```ts
const config = {
  defaultSeverity: 'error',
  jsRules: {},
  rules: {
    // stuff
  }
};

linter.lint(fileName, fileContents, config);
```

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
